### DISCUSSION TITLE
Continuous delivery of dashboards

### DISCUSSION THREAD

**Discussion by danielgauci** [datalust/seq-tickets#1571] [Help]
Hi guys,

Amazing product. Have been using it for 6+ years and it continues to exceed my expectations. I can't imagine supporting any  product small or large without Seq :)

Wondering if you could provide some direction or guidance on below. For context we run Seq on local dev environments and 1 Seq instance per cloud environment (staging + production) via Helm charts.

Continuous delivery of dashboards / signals / metadata between Seq instances/environments. We keep our local environment dashboards in sync with staging and production dashboards which helps with debugging cloud environment issues on our locals a breeze. I can't see how this is easily achievable at the moment. For example, exporting local dev dashboards/signals as JSON into Git and deploying these with our infrastructure releases alongside the Grafana dashboards without the need for all Seq instances to share the same secret keys.

Dashboards start in Git and make there way through the cloud environments staging and then production. Additionally if a dashboard could be marked as "read only" to prevent confusion when it has been deployed and overwritten would be awesome. Similar to Grafana's "read only" flag on dashboards.

Managing the above for two Seq instances is okay for now but as we expand our production environment to multi-region managing the above is going to be a challenge.

I may have missed information in the current docs if so please direct me to the correct doc :)

Thanks in advance! 
Dan

**Comment by nblumhardt** [comment]
Thanks for dropping by, Dan!

Is https://blog.datalust.co/copying-signals-and-dashboards-using-seqcli-templates/ close to what you're looking for? There might still be some gaps in the template export/import/merge story, let us know if so.

Hope this helps,
Nick

**Reply by danielgauci** [reply]
Amazing! 

Thanks so much mate. 

Would be great to scope the import and export commands to a single workspace including all the dashboards under the workspace. Use case - running multiple solutions / client project on local machine and only wanting to export a single workspace inc dashboards. Will use separate Seq instances on local for now though no biggie.

**Reply by nblumhardt** [reply]
Great 👍 thanks Daniel.

I think adding a `seqcli workspace show` command capable of listing signal, dashboard, and query ids in a workspace, would make automating your scenario easier - will give it a thought!

Using raw HTTP to `GET` the workspace and iterate through its included dashboard ids might be doable, if you're using a shell with JSON support, as a workaround 🤔
